### PR TITLE
Jennifer/add profile editing

### LINKED ISSUE #18: [Profile] Profile Editing (FE-2)
Profile Editing

### Acceptance Criteria
- [x] Add "Edit Profile" button on profile page (only visible on own profile)
- [x] Redirect to `/onboarding` route with existing data pre-filled
  - [x] Pre-fill welcome page (name, age_group)
  - [x] Pre-fill interests page (categories, hobbies, avoid)
  - [x] Pre-fill style compass (all slider values)
- [x] Update onboarding flow to handle edit mode
  - [x] Use PATCH
  - [x] Redirect back to profile page on completion

### PR DESCRIPTION
**Pull request by jennifertieu**
Closes #18 

- [x] Add "Edit Profile" button on profile page (only visible on own profile)
- [x] Redirect to `/onboarding` route with existing data pre-filled
  - [x] Pre-fill welcome page (name, age_group)
  - [x] Pre-fill interests page (categories, hobbies, avoid)
  - [x] Pre-fill style compass (all slider values)
- [x] Update onboarding flow to handle edit mode
  - [x] Use PATCH
  - [x] Redirect back to profile page on completion
  
I had to refactor the middleware to be able to redirect to `/onboarding`

### CODE REVIEW

File: components/ProfileCard/ProfileCard.tsx [hunk 1]
```diff
@@ -5,13 +5,20 @@ import Avatar from "@/components/Avatar/Avatar";
...
   profile: Profile | null;
   showEditButton?: boolean;
 }
 
 const ProfileCard = ({ profile, showEditButton = false }: ProfileCardProps) => {
+  const router = useRouter();
+
+  const handleEditProfile = () => {
+    router.push("/onboarding?editing=true");
+  };
```
Comment (thomasnguyen) [inline components/ProfileCard/ProfileCard.tsx:20]: Nice

**Review by thomasnguyen — APPROVED** [review]
lgtm

**Comment by thomasnguyen** [comment]
Oh can you check out and fix the build failures before merging?

**Review by kepsteen — APPROVED** [review]
LGTM once the build passes

**Pull request merged**